game-core/src/main/java/games/strategy/engine/chat/LobbyChatTransmitter.java

```diff
@@ -21,30 +28,47 @@
 
   @Override
   public void setChatClient(final ChatClient chatClient) {
-    playerToLobbyConnection.addChatMessageListeners(
-        ChatMessageListeners.builder()
-            .playerStatusListener(chatClient::statusUpdated)
-            .playerLeftListener(chatClient::participantRemoved)
-            .playerJoinedListener(chatClient::participantAdded)
-            .chatMessageListener(chatClient::messageReceived)
-            .connectedListener(chatClient::connected)
-            .chatEventListener(chatClient::eventReceived)
-            .playerSlappedListener(
-                slapEvent -> {
-                  if (slapEvent.getSlapped().equals(localUserName)) {
-                    chatClient.slappedBy(slapEvent.getSlapper());
-                  } else {
-                    chatClient.playerSlapped(
-                        slapEvent.getSlapper() + " slapped " + slapEvent.getSlapped());
-                  }
-                })
-            .serverErrorListener(log::severe)
-            .build());
+    playerToLobbyConnection.addMessageListener(
+        ChatterListingMessage.TYPE, message -> chatClient.connected(message.getChatters()));
+
+    playerToLobbyConnection.addMessageListener(
+        PlayerStatusUpdateReceivedMessage.TYPE,
+        message -> chatClient.statusUpdated(message.getUserName(), message.getStatus()));
+
+    playerToLobbyConnection.addMessageListener(
+        PlayerLeftMessage.TYPE, message -> chatClient.participantRemoved(message.getUserName()));
+
+    playerToLobbyConnection.addMessageListener(
+        PlayerJoinedMessage.TYPE,
+        message -> chatClient.participantAdded(message.getChatParticipant()));
+
+    playerToLobbyConnection.addMessageListener(
+        ChatReceivedMessage.TYPE,
+        message -> chatClient.messageReceived(message.getSender(), message.getMessage()));
+
+    playerToLobbyConnection.addMessageListener(
+        PlayerJoinedMessage.TYPE,
+        message -> chatClient.participantAdded(message.getChatParticipant()));
+
+    playerToLobbyConnection.addMessageListener(
+        ChatEventReceivedMessage.TYPE, message -> chatClient.eventReceived(message.getMessage()));
+
+    playerToLobbyConnection.addMessageListener(
+        PlayerSlapReceivedMessage.TYPE,
+        message -> {
+          if (message.getSlappedPlayer().equals(localUserName)) {
+            chatClient.slappedBy(message.getSlappingPlayer());
+          } else {
+            chatClient.eventReceived(
+                message.getSlappingPlayer() + " slapped " + message.getSlappedPlayer());
+          }
+        });
+    playerToLobbyConnection.sendConnectToChatMessage();
   }
 
   @Override
   public Collection<ChatParticipant> connect() {
-    return playerToLobbyConnection.connect();
+    return List.of();
```

Comment:
Certainly. Before we had more of a hybrid http and websocket model, sync and async. On the very first iteration we had a problem where if you initially connect to websocket, we would not know the chatter name. So what we did was the initial connection established a chatter, and we got the list of chatters back in a synchronized manner.

This is being updated so that we establish a websocket session first, but even with the websocket session you are not going to get any messages until you send a 'join' message. With that, the server will respond to 'join' messages with a websocket response containing the chatter list.